L123-core-add-alts-google-call-credentials.md

```diff
@@ -0,0 +1,83 @@
+# Support ALTS Credentials in Google Default Credentials
+
+--------------------------------------------------------------------------------
+
+*   Author(s): anniefrchz
+*   Approver: a11r
+*   Status: Draft
+*   Implemented in: C++
+*   Last updated: 2025/06/30
+*   Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+This proposal outlines a change to the gRPC Core C-API to support
+alts-credentials configurations within `grpc_google_default_credentials`. This
+enhancement specifically allows a secondary set of channel credentials for ALTS
+to be provided alongside the default credentials.
+
+### Background
+
+The existing `grpc_google_default_credentials_create` function only allowed for
+the configuration of a single set of call credentials. This posed a limitation
+in scenarios where a client needs to communicate with different services that
+require distinct security mechanisms over the same channel. Without the ability
+to handle both credential types, this process would be cumbersome or impossible
+to manage on a single channel. This proposal addresses the need to support
+multiple transport security types on a channel initialized with Google's default
+credentials.
+
+### Proposal
+
+To maintain backward compatibility with the existing C-API, this proposal
+modifies the function `grpc_google_default_credentials_create` to add a second
+set of call credentials.
+
+```c
+GRPCAPI grpc_channel_credentials* grpc_google_default_credentials_create(
+    grpc_call_credentials* tls_credentials,
+    grpc_call_credentials* alts_credentials);
+```
+
+This new function accepts two arguments:
+
+1.  `tls_credentials`: The primary call credentials, consistent with the
+    existing API. Usually. default back to the TLS connection.
+2.  `alts_credentials`: A secondary set of channel credentials to be used
+    specifically for ALTS connections.
+
+When a channel is created with this function, the gRPC runtime will select the
+appropriate credentials based on the transport security type required for a
```

Comment:
The transport security type is populated after the secure connection is established and indicates the channel type (ie, TLS or ALTS). It's not a requirement. Please reword this paragraph.